### ISSUE TITLE
abstract class override fields

### ISSUE THREAD

**Issue by hadukin** [passsy/dart-lint#43]
I think these 2 errors contradict each other

lint: ^1.6.0

`
abstract class IContent {
  final String? name;
  IContent({
    this.name,
  });
}

class Content extends IContent {
  // 2 conflicting errors
  // Annotate overridden members.dart(annotate_overrides)
  // Don't override fields.dart(overridden_fields)
  final String? name;
}
`

**Comment by minhqdao** [comment]
The errors seem to contradict, but in fact they don't. 😉

The first one states that anything carrying the same name as a counterpart in its super class should be annotated with `@override`. That is typically the case if you define a `static` method in an `abstract` class that another class implements/extends. You could also do that here to resolve the first error. But then there is the second error stating that, no matter if `@override` is added or not, you shouldn't do it in the first place because fields aren't meant to be overridden.

See, if `IContent` had a field called `name` and `Content` had `firstName`, you could access both through:

```dart
final content = Content();
final name = content.name;
final firstName = content.firstName;
```

because `Content` extends `IContent` with its inherited `name` field. If you override a field with another, you loose that functionality, which, however, is the entire purpose of extending it. It is therefore [bad practice](https://dart-lang.github.io/linter/lints/overridden_fields.html) to do so. 😔

If you really want to enforce the override, you can achieve that with an `// ignore ...`, but maybe you want to reevaluate whether you want to do that in the first place. 🤔
